### PR TITLE
Add support for filtered/public configuration

### PR DESCRIPTION
**Pull request by beck**
I've included some samples in the tests dir.  This goes against the grain with what I've seen in with the rest of the tests, where test fixtures are generated and destroyed.  IMO it is much easier to understand how configurator works when there are tangible examples.

As for the implementation, I decided to go with a file called `public-data.json` in app config.  This information is totally different and unlike the other config so I thought it appropriate if the file unlike the others.

ping @stefanor

closes https://github.com/yola/yoconfigurator/issues/28

### CODE REVIEW

**Comment by beck** [comment]
:bell:

File: yoconfigurator/base.py [hunk 1]
```diff
@@ -11,13 +13,32 @@ def default(self, obj):
...
+def _write(content, name):
+    with open(name, 'w') as f:
+        json.dump(
+            content, f, indent=2, cls=DetectMissingEncoder,
+            separators=(',', ': '))
+
+
+def write_configs(app_dir, configs):
+    """Write configurations to the applicaiton directory."""
+    for config in configs:
```
Comment (stefanor) [inline yoconfigurator/base.py]: I think given only two calls, they are prettier.

We aren't expecting to be writing many files, so this is over-engineering the code, at a cost to the API, for no real  gain.

Comment (beck) [inline yoconfigurator/base.py]: updated

File: yoconfigurator/dicts.py [hunk 2]
```diff
@@ -116,3 +127,11 @@ def merge_dicts(d1, d2, _path=None):
         raise TypeError('Cannot merge a %s with a %s' % (type(d1), type(d2)))
 
     return d1
+
+
+def filter_dict(unfiltered, filter_keys):
+    '''Return a subset of a dictionary using the specified keys.'''
```
Comment (thomasw) [inline yoconfigurator/dicts.py]: `"""` not `'''`: https://www.python.org/dev/peps/pep-0257/

Comment (beck) [inline yoconfigurator/dicts.py]: I know... but I was sticking with the convention used throughout the repo.

Comment (beck) [inline yoconfigurator/dicts.py]: will update.

**Comment by beck** [comment]
Cleaned up docstrings and added a some documentation to the readme.

File: README.md [hunk 3]
```diff
@@ -19,10 +19,26 @@ from the application.
...
   ```
-  python -m unittest -v yoconfigurator.tests
+  python -m unittest discover
   ```
 
 
+## Public configuration
+
+If it is necessary to share configuration with non-privledged sources
+(example: publishing API urls), a publicly consumable configuration is required.
```
Comment (stefanor) [inline README.md]: example: publishing API URLs, but not related credentials?

**Comment by stefanor** [comment]
:+1: very happy with the cleanup, and the final look of this.

**Pull request merged**